jaxrs-clients/src/main/java/com/palantir/remoting3/jaxrs/AbstractFeignJaxRsClientBuilder.java

```diff
@@ -62,19 +62,26 @@
 abstract class AbstractFeignJaxRsClientBuilder {
 
     private final ClientConfiguration config;
+    /**
+     * The primary URI used to bootstrap the Feign client; this is the URI used by Feign to create an OkHttp call. Note
+     * that the {@link MultiServerRetryInterceptor} replaces this URI with a random URI from the client configuration
+     * before making the request.
+     */
+    private final String primaryUri;
 
     AbstractFeignJaxRsClientBuilder(ClientConfiguration config) {
-        this.config = config;
         Preconditions.checkArgument(config.maxNumRetries() == 0,
                 "Connection-level retries are not supported by %s", JaxRsClient.class.getSimpleName());
+        Preconditions.checkArgument(!config.uris().isEmpty(), "Must provide at least one service URI");
```

Comment:
don't think we can require this as per your earlier email saying that initial uri lists may be empty